src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java

```diff
@@ -151,7 +152,10 @@ public static void processEntityDeathEvent(EntityDeathEvent event) {
      * @param event the event - this event is only called for Players (not entities)
      */
     public static void processBlockBreakEvent(BlockBreakEvent event) {
-        if(!TownyResourcesSettings.areResourceExtractionLimitsEnabled())
+        if(!TownyResourcesSettings.areResourceExtractionLimitsEnabled()
+                || BypassEntries.bypassData.contains(event.getPlayer().getUniqueId())
+                || event.getPlayer().hasPermission("townyresources.bypass")
+        )
```

Comment:
The three similar tests added to this class can be extracted to a `private boolean isPlayerExtractLimited(Player player)` method.